### PR TITLE
Triangle: Update tests to check properties, not types

### PR DESCRIPTION
**Pull request by IanWhitney**
As discussed here: https://github.com/exercism/x-common/issues/379

Converts the tests so that they are checking the property of a triangle
and not its type (or type stand-in like symbol/atom/etc.).

I've removed some duplicative tests, but I believe I've still covered
all the edge cases that were part of the old test suite.

### CODE REVIEW

File: exercises/triangle/canonical-data.json [hunk 1]
```diff
@@ -8,88 +8,115 @@
...
+              "sides": [0, 0, 0],
+              "expected": false
         },
         {
-            "description": "isosceles triangle with first and last sides equal",
-            "sides": [4, 3, 4],
-            "expected": "isosceles"
+              "#": "Your track may choose to skip this test and deal only with integers if appropriate",
+              "description": "sides may be floats",
+              "sides": [0.5, 0.4, 0.6],
```
Comment (Insti) [inline exercises/triangle/canonical-data.json]: This doesn't look equilateral.

File: exercises/triangle/canonical-data.json [hunk 2]
```diff
@@ -8,88 +8,115 @@
...
+              "sides": [4, 4, 4],
+              "expected": true
         },
         {
-            "description": "a triangle violating the triangle inequality is illegal",
-            "sides": [7, 3, 2],
-            "expected": "illegal"
+              "description": "false if two sides are equal",
+              "sides": [4, 4, 3],
+              "expected": true
```
Comment (Insti) [inline exercises/triangle/canonical-data.json]: true != false

File: exercises/triangle/canonical-data.json [hunk 3]
```diff
@@ -8,88 +8,115 @@
...
+              "sides": [4, 3, 4],
+              "expected": true
         },
         {
-            "description": "2a == b+c looks like equilateral, but isn't always",
-            "sides": [5, 4, 6],
-            "expected": "scalene"
+              "description": "true if three sides are equal",
+              "sides": [4, 4, 4],
+              "expected": true
```
Comment (Insti) [inline exercises/triangle/canonical-data.json]: Might be good to add a comment or expansion of the description here that clarifies that this is BOTH isosceles and equilateral.
